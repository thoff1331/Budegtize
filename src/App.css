body {
  background-color: rgb(12, 182, 173);
}

.app-container {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
  height: 100vh;
  width: 100vw;
}
.user-elements {
  width: 25vw;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.user-elements input {
  width: 20vw;
  border: none;
  height: 5vh;
  text-align: center;
  outline: none;
  border-radius: 10%;
  background-color: rgb(12, 182, 173);
  font-weight: bolder;
  color: black;
  font-size: 30px;
}

.converted-number {
  display: flex;
  justify-content: center;
  height: 10vh;
  color: rgb(248, 101, 2);
  font-size: 70px;
}

.user-elements button {
  height: 100px;
  width: 100px;
  border-radius: 50%;
  justify-content: center;
  color: black;
  font-size: 20px;
  font-weight: 900;
  border: 4px solid black;
  background-color: white;
  outline: none;
}
