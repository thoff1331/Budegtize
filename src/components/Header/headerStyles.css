.header-container {
  display: flex;
  justify-content: space-between;
  flex-direction: row;
}

.header {
  display: flex;
  justify-content: space-around;
  align-items: center;
  flex-direction: row;
  height: 10vh;
  width: 100vw;
  background-color: rgb(15, 146, 146);
  color: rgb(238, 218, 218);
}
