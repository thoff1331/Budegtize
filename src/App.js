import React, { Component } from "react";
import axios from "axios";
import "./App.css";

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      api: "",
      apiInput: "",
    };
  }
  handleChange = (event) => {
    this.setState({ [event.target.name]: event.target.value });
  };

  handleApi = (e, num) => {
    if (this.state.apiInput > 255) {
      alert("Please Enter a number between 1 & 255");
    } else {
      axios
        .get("/api/num", {
          params: {
            integer: this.state.apiInput,
          },
        })
        .then((response) => {
          this.setState({
            api: response.data,
          });
        });
    }
  };
  render() {
    return (
      <div className="app-container">
        <h1>Enter a Number to convert it to a Roman Numeral</h1>
        <div className="user-elements">
          <input
            placeholder="Enter a number"
            type="number"
            onChange={this.handleChange}
            name="apiInput"
          />
          <button
            id="convert-button"
            onClick={(e) => this.handleApi(e, this.state.apiInput)}
          >
            Convert
          </button>
        </div>
        <h1 className="converted-number">
          {" "}
          Your Number as A Roman Numeral is :{this.state.api}
        </h1>
      </div>
    );
  }
}

export default App;
