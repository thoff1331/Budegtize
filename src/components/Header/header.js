import "./headerStyles.css";

function Header() {
  return (
    <div className="header-container">
      <div className="header">
        <h1>Budgetize</h1>
        <h4>My Budget</h4>
        <h4>Stocks</h4>
        <h4>My stocks</h4>
      </div>
    </div>
  );
}

export default Header;
