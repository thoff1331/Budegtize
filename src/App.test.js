import { shallow } from "enzyme";
import App from "./App";
import Header from "./components/Header/header";
it("renders without crashing", () => {
  shallow(<App />);
});

it("haeder renders without crashing", () => {
  shallow(<Header />);
});

it("renders app header", () => {
  const wrapper = shallow(<Header />);
  let welcome = <h1>Budgetize</h1>;

  expect(wrapper.contains(welcome)).toEqual(true);
});
