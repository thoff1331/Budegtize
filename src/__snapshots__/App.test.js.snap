// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should pass a selected value to the onChange function 1`] = `
<div
  className="app-container"
>
  <h1>
    Enter a Number to convert it to a Roman Numeral
  </h1>
  <div
    className="user-elements"
  >
    <input
      name="userInput"
      onChange={[Function]}
      placeholder="Enter a number"
      type="number"
    />
    <button
      id="convert-button"
      onClick={[Function]}
    >
      Convert
    </button>
  </div>
  <h1
    className="converted-number"
  >
    Your Number as A Roman Numeral is :
  </h1>
</div>
`;
